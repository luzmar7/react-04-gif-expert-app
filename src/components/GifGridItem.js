import React from 'react'


const GifGridItem = ( { id, title, url } ) => {
 
  return (
    <div className='card animate__animated animate__fadeIn'>
      {/* { img.title } */}
      <img src= { url  } alt={ title }/>
      <p>{ title }</p>
      
    </div>
  )
}

export default GifGridItem
