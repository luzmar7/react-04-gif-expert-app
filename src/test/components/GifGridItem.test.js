import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'

import GifGridItem from '../../components/GifGridItem'


describe('Pruebas en <GifGridItem />', () => {

  const title = 'Un titulo'
  const url = 'https://github.com/luzmar7/react-03-counter-app/blob/master/src/tests/CounterApp.test.js'
  
  let wrapper = shallow( <GifGridItem title = { title} url =  { url }/>)
  beforeEach(() =>{
    wrapper = shallow( <GifGridItem title = { title} url =  { url }/>)
  })

  test('debe mostrar <GifGridItem /> correctamente', () => {
    wrapper = shallow( <GifGridItem title = { title} url =  { url }/>)
    expect( wrapper ).toMatchSnapshot()
    
  })

  test('debe tener un parrafo con el title', () => {

    const p = wrapper.find('p')
    expect(p.text().trim()).toBe( title )
    
  })
  test('debe de tener la imagen igual al url y alt de los props', () => {
    const img = wrapper.find('img')
    // console.log(img.props())
    expect(img.prop('src')).toBe(url)
    expect(img.prop('alt')).toBe(title)
  })

  test('debe tener animate__fadeIn', () => {
    const div = wrapper.find('div')
    // animate__fadeIn
    const className = div.prop('className')
    expect(className.includes('animate__fadeIn')).toBe(true)

    
  })
  


  
  





  
})
