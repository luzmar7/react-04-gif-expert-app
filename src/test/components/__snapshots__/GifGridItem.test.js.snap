// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://github.com/luzmar7/react-03-counter-app/blob/master/src/tests/CounterApp.test.js"
  />
  <p>
    Un titulo
  </p>
</div>
`;
